lib/inspec/utils/install_context.rb

```diff
@@ -0,0 +1,55 @@
+module Inspec
+
+  # Hueristics to determine how InSpec was installed.
+  module InstallContextHelpers
+
+    def guess_install_context
+      # These all work by simple path recognition
+      return "chef-workstation" if chef_workstation_install?
+      return "omnibus" if omnibus_install?
+      return "chefdk" if chefdk_install?
+      return "habitat" if habitat_install?
+
+      # Order matters here - gem mode is easily mistaken for one of the above
+      return "docker" if docker_install?
+      return "rubygem" if rubygem_install?
+
+      return "source" if source_install?
+
+      "unknown"
+    end
+
+    def chef_workstation_install?
+      !!(src_root.start_with?("/opt/chef-workstation") || src_root.start_with?("C:/opscode/chef-workstation"))
+    end
+
+    def chefdk_install?
+      !!(src_root.start_with?("/opt/chef-dk") || src_root.start_with?("C:/opscode/chef-dk"))
+    end
+
+    def docker_install?
+      # Our docker image is based on alpine. This could be easily fooled.
+      !!(rubygem_install? && File.exist?("/etc/alpine-release"))
+    end
+
+    def habitat_install?
+      !!src_root.match(%r{hab/pkgs/chef/inspec/\d+\.\d+\.\d+/\d{14}})
+    end
+
+    def omnibus_install?
+      !!(src_root.start_with?("/opt/inspec") || src_root.start_with?("C:/opscode/inspec"))
+    end
+
+    def rubygem_install?
+      !!src_root.match(%r{gems/inspec-\d+\.\d+\.\d+})
```

Comment:
I may need this explained to me.

When vendored, Bundler makes a directory `vendor/cache` full of gems - but the gems are not yet installed/expanded.  Presumably when Bundler / Rubygems activates a gem it then installs the gem somewhere? So... wouldn't the above regex work?

Reply:
I think when vendored, they're installed under `vendor` as well... Our BK "bundle install" now takes <20s because we've untarred the vendor directory from cache.